src/main/java/org/sunbird/digilocker/service/DigiLockerIntegrationServiceImpl.java

```diff
@@ -0,0 +1,243 @@
+        if (MapUtils.isNotEmpty(digiLockerDocInfo)) {
+            String content = getCertificate((String) digiLockerDocInfo.get(Constants.CERTIFICATE_ID));
+            if (StringUtils.isNotEmpty(content)) {
+                encodeToFile(content);
+                docDetails.setDocContent(content.toString());
+                response.setResponseStatus(new PullDocResponse.ResponseStatus(1, new Date().toString(), request.getTxn()));
+            } else {
+                response.setResponseStatus(new PullDocResponse.ResponseStatus(0, new Date().toString(), request.getTxn()));
+            }
+            PullDocResponse.CertificateInfo certificateInfo = new PullDocResponse.CertificateInfo();
+            certificateInfo.setCertificateName((String)digiLockerDocInfo.get(Constants.CERTIFICATE_NAME));
+            certificateInfo.setIssuedOn(dateFormat.format(digiLockerDocInfo.get(Constants.LAST_ISSUED_ON)));
+            docDetails.setDataContent(certificateInfo);
+            response.setDocDetails(docDetails);
+        } else {
+            response.setResponseStatus(new PullDocResponse.ResponseStatus(0, new Date().toString(), request.getTxn()));
+        }
+        return response;
+    }
+
+    private String getCertificate(String certificateId) {
+        Map<String, Object> response = (Map<String, Object>) outboundRequestHandlerService.fetchResult(serverProperties.getCertRegistryServiceBaseUrl() +
+                serverProperties.getCertRegistryCertificateDownloadUrl() + certificateId);
+        if (null != response && Constants.OK.equalsIgnoreCase((String) response.get(Constants.RESPONSE_CODE))) {
+            Map<String, Object> certificateInfo = (Map<String, Object>) response.get(Constants.RESULT);
+            if (MapUtils.isNotEmpty(certificateInfo)) {
+                byte[] out = null;
+                try {
+                    out = generatePdfFromSvg(decodeUrl((String) certificateInfo.get("printUri")));
+                    return encodeBytesToBase64(out);
+                } catch (Exception e) {
+                    logger.error("" + e);
+                }
+            }
+        }
+        return null;
+    }
+
+    private String encodeBytesToBase64(byte[] bytes) {
+        return Base64.getEncoder().encodeToString(bytes);
+    }
+
+    public byte[] generatePdfFromSvg(String svgContent) throws IOException, TranscoderException {
+        String svgContentDecode = svgContent.replace("data:image/svg+xml,", "");
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            TranscoderInput input = new TranscoderInput(new ByteArrayInputStream(svgContentDecode.getBytes()));
+            TranscoderOutput output = new TranscoderOutput(outputStream);
+            PDFTranscoder transcoder = new PDFTranscoder();
+            transcoder.transcode(input, output);
+            return outputStream.toByteArray();
+        }
+    }
+
+    private void encodeToFile(String encodedString) {
```

Comment:
filePath -- do we have to attach empty pdf ??
if so, do add it as resource and read the same.

Reply:
Removed this used this for testing purpose